src/ObjectList.php

```diff
@@ -362,6 +363,60 @@ private function permute(&$out, $list, $perms = [], $max = null)
     }
 
     //
+    // Serialization and deserialization
+    //
+
+    /**
+     * Turn this object into an array using toArray method on each element if they have it.
+     *
+     * @return array
+     */
+    public function toArray()
+    {
+        $out = [];
+        foreach ($this->getArrayCopy() as $key => $value) {
+            if (is_object($value) && method_exists($value, 'toArray')) {
```

Comment:
Well, it's possible for objects not to be of the same class (you can validate input by interface, or not at all, if you so choose).